### PR TITLE
Add com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0

### PR DESCRIPTION
**Pull request by BenFradet**
I would need a fallback case so that if you want to run a custom not supported out of the box recovery scenario you don't need to update the schema, wdyt would be the best way to go about it?

### CODE REVIEW

File: schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    "anyOf": [
+      {
+        "properties": {
+          "name": {
+            "description": "The PassThrough recovery scenario doesn't modify the payload",
+            "enum": [
+              "PassThrough"
+            ]
+          },
+          "error": {
```
Comment (chuwy) [inline schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: Should we mention format of `error`s in the description?

File: schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0 [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        "required": [
+          "name",
+          "error",
+          "toReplace",
+          "replacement"
+        ],
+        "additionalProperties": false
+      },
+      {
+        "properties": {
```
Comment (chuwy) [inline schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: I'd still prefer an option with self-describing JSON as a fallback. `additionalProperties` makes this fallback effectively untyped.

Comment (BenFradet) [reply schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: The point is not to have to upload a schema to run a custom recovery scenario, I think having a self-desc fallback defeats this purpose, wdyt?

File: schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0 (hunk 1 above)
Comment (BenFradet) [reply schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: wdym by format?

File: schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0 (hunk 2 above)
Comment (chuwy) [reply schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: At one hand - yes, definitely, but on the other hand if somebody wrote a custom recovery scenario - uploading a schema would be a least of their problems. At the same time schema'ed recovery can motivate one to publish whole recovery scenario.

Also, even if recovery will be written in Scala - JSON schema has stronger validation, so quite a lot invalid configs can be filtered out.

File: schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0 (hunk 1 above)
Comment (chuwy) [reply schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: Sorry I though its going to contain complex logic such as `contains("invalid schema") && contains("com.acme")`, but apparently this is going to be simple matcher.

File: schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0 (hunk 2 above)
Comment (BenFradet) [reply schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0]: I'm not quite sure adding steps is going to encourage contributions but anyways the main problem I'm seeing with this approach is that I'm not going to have the same ADT in the `data` array, that seems like unnecessary complexity, wdyt?

**Comment by alexanderdean** [comment]
Can we just move this back to R100 @BenFradet, it is blocking R99 release, cc @yalisassoon

**Comment by BenFradet** [comment]
sure

**Comment by alexanderdean** [comment]
Pushed back, unblocks R99

**Comment by alexanderdean** [comment]
@BenFradet if you want to expedite this I'm sure the other stuff can be moved out of R100

**Comment by BenFradet** [comment]
yes will do, I will just wait for Anton to get back

**Pull request closed without merging**